crates/starknet-devnet-core/src/starknet/add_declare_transaction.rs

```diff
@@ -100,8 +107,12 @@ pub fn add_declare_transaction_v1(
         .calculate_transaction_hash(&starknet.config.chain_id.to_felt(), &class_hash)?;
 
     let declare_transaction =
-        broadcasted_declare_transaction.create_declare(class_hash, transaction_hash);
-    let transaction = Transaction::Declare(DeclareTransaction::Version1(declare_transaction));
+        DeclareTransactionV0V1::new(&broadcasted_declare_transaction, class_hash);
+
+    let transaction = TransactionWithHash::new(
+        transaction_hash,
+        Transaction::Declare(DeclareTransaction::Version1(declare_transaction)),
```

Comment:
Why not rename `Version1` to `V1` etc.? It is less verbose but conveys the same information, and maybe it will lead to more line count reduction after re-formatting.

Reply:
I didnt think of that, my idea was not to change the enum variants. But definitely will do it in this PR